pallets/settlement/src/benchmarking.rs

```diff
@@ -0,0 +1,568 @@
+    }
+}
+
+fn populate_legs_for_instruction<T: Trait>(index: u32, legs: &mut Vec<Leg<T::Balance>>) {
+    let ticker = Ticker::try_from(vec![b'A'; index as usize].as_slice()).unwrap();
+    legs.push(Leg {
+        from: generate_portfolio::<T>("from_did", index, 100, None),
+        to: generate_portfolio::<T>("to_did", index, 500, None),
+        asset: ticker,
+        amount: 100.into(),
+    });
+}
+
+fn get_settlement_type<T: Trait>(l: u32) -> SettlementType<T::BlockNumber> {
+    match l {
+        0 => SettlementType::SettleOnAffirmation,
+        _ => {
+            set_block_number::<T>(50);
+            SettlementType::SettleOnBlock(100.into())
+        }
+    }
+}
+
+benchmarks! {
+    _{}
+
+    create_venue {
+        // Variations for the venue_details length.
+        let d in 1 .. MAX_VENUE_DETAILS_LENGTH;
+        // Variations for the no. of signers allowed.
+        let s in 0 .. MAX_SIGNERS_ALLOWED;
+        let mut signers = Vec::with_capacity(s as usize);
+        let Account {origin, did, .. } = make_account::<T>("caller", SEED);
+        let venue_details = VenueDetails::from(vec![b'D'; d as usize].as_slice());
+        let venue_type = VenueType::Distribution;
+        // Create signers vector.
+        for signer in 0 .. s {
+            signers.push(make_account::<T>("signers", signer).account_id);
+        }
+    }: _(origin, venue_details, signers, venue_type)
+    verify {
+        ensure!(matches!(Module::<T>::venue_counter(), 2), "Invalid venue counter");
+        ensure!(matches!(Module::<T>::user_venues(did).into_iter().last(), Some(1)), "Invalid venue id");
+        ensure!(Module::<T>::venue_info(1).is_some(), "Incorrect venue info set");
+    }
+
+
+    update_venue {
+        // Variations for the venue_details length.
+        let d in 1 .. MAX_VENUE_DETAILS_LENGTH;
+        let venue_details = VenueDetails::from(vec![b'D'; d as usize].as_slice());
+        // Venue type.
+        let venue_type = VenueType::Sto;
+        let Account {account_id, origin, did} = make_account::<T>("creator", SEED);
+        // create venue
+        let venue_id = create_venue_::<T>(did, vec![]);
+    }: _(origin, venue_id, Some(venue_details), Some(venue_type))
+    verify {
+        let updated_venue_details = Module::<T>::venue_info(1).unwrap();
+        ensure!(matches!(updated_venue_details.venue_type, VenueType::Sto), "Incorrect venue type value");
+        ensure!(matches!(updated_venue_details.details, venue_details), "Incorrect venue details");
+    }
+
+
+    add_instruction {
+
+        let l in 1 .. T::MaxLegsInInstruction::get() as u32; // Variation for the MAX leg count.
+        let mut legs = Vec::with_capacity(l as usize);
+
+        // create venue
+        let Account {account_id, origin, did} = make_account::<T>("creator", SEED);
+        let venue_id = create_venue_::<T>(did, vec![]);
+
+        // Define settlement type
+        let settlement_type = get_settlement_type::<T>(l % 2);
```

Comment:
This poisons the benchmark. Please create two separate benchmarks for different settlement types.

Reply:
How does it poison the benchmark?

Reply:
The benchmark tries to do regression analysis on the input variables. You can only do that if everything other than the variable being analyzed in constant. You are adding a mother variable based on the input variable which renders the data not useable for regression analysis. I can go into details over a quick call, that'll be easier.